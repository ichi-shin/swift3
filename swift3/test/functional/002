#!/bin/bash

# PUT Bucket

. ./common

echo '
<CreateBucketConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
<LocationConstraint>US</LocationConstraint>
</CreateBucketConfiguration>' > $tmp.default.xml

echo '
<CreateBucketConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
<LocationConstraint>us-west-2</LocationConstraint>
</CreateBucketConfiguration>' > $tmp.us_west_2.xml

S3USER=tester

_s3_put /bucket -D - -T $tmp.default.xml | _filter_curl
_s3_put /bucket -D - -T $tmp.us_west_2.xml | _filter_curl xml
