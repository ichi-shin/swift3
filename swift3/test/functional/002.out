QA output created by 002
> s3curl --id tester -- -X PUT -D - -T /TMP.default.xml http://SWIFT_HOST/bucket... 200
HTTP/1.1 100 Continue

HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
Location: /bucket
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

> s3curl --id tester -- -X PUT -D - -T /TMP.us_west_2.xml http://SWIFT_HOST/bucket... 400
HTTP/1.1 100 Continue

HTTP/1.1 400 Bad Request
Content-Type: text/xml
Date: DATE
Transfer-Encoding: chunked
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<Error>
  <Code>InvalidLocationConstraint</Code>
  <Message>The specified location constraint is not valid.</Message>
  <RequestId>TXID</RequestId>
</Error>
